implementation/src/main/java/io/smallrye/mutiny/Uni.java

```diff
@@ -389,6 +393,25 @@ static UniCombine combine() {
      */
     Uni<T> cache();
 
+    // TODO
+    Uni<T> cacheUntil(BooleanSupplier invalidationGuard);
+
+    // TODO
+    default Uni<T> cacheFor(Duration duration) {
+        return cacheUntil(new BooleanSupplier() {
+            private volatile long startTime = System.currentTimeMillis();
```

Comment:
As soon as an event gets invalid there will be a new upstream subscription on every new subscription, did I get it right?

As a user I would expect, that a newly received item will also be cached for that duration.

Some kind of "reset start time" triggered by the UniCache after resubscription as implementation idea?
Or more functional: a BooleanSupplierSupplier, called on first and every further resubscription...

Reply:
Here it caches for `subscription_time + duration`. On the next invalidation it triggers a new upstream subscription, and counts from here again.

Reply:
See the tests

Reply:
@heubeck I've updated this to start counting at the first subscription rather than at the  operator creation.